### ISSUE TITLE
Curve alt_bn_128

### ISSUE THREAD

**Issue by darioAnongba** [indutny/elliptic#154]
Hi,

I am currently heavily involved in Ethereum development and am currently implementing a Confidential Transactions token. The curve used for this is alt_bn_128 and not secp256k1 like in Bitcoin. The curve [alt_bn_128](https://eips.ethereum.org/EIPS/eip-196) is similar to secp256k1 but wiht parameters:
- A = 0
- B = 3
- P = 0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47
- N = 0x30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001

I was wondering if defining a new curve similar to secp256k1 with those parameters would "work out of the box" or if it's not that trivial. If not, what should I do to add support for this curve?

Also, I couldn't find documentation anywhere on how to use this library and its API (list of functions, etc) Does it exist?

Thanks.

**Comment by mahrud** [comment]
I don't think so, given that it's a Barreto-Naehrig curve, not a Koblitz curve like secp256k1. It'll most likely require a new curve type to be defined.

Also, I don't know much about Ethereum, but I presume what you really want to do with the curves is more than just arithmetic operations and making signatures. In particular, I think you really want to use elliptic pairings, which definitely requires some work.

Re: documentation, I think the README is the only thing AFAIK.

**Comment by darioAnongba** [comment]
It would actually be very useful for the Ethereum community to be able to use elliptic pairings in JavaScript but it is not for me at the moment. I actually just need the arithmetic operations and not the pairings.

You surely are asking yourself why I use alt_bn_128 if I don't need pairings. It's because the cost of using the alt_bn_128 curve will surely be greatly decreased because of the EIP 196 and 197 : https://eips.ethereum.org/EIPS/eip-196 and https://eips.ethereum.org/EIPS/eip-197. 

Ethereum is famous and used enough to justify officially adding the curve to this library in the future
